tools/linter/adapters/pip_init.py

```diff
@@ -52,13 +55,20 @@ def run_command(args: list[str]) -> subprocess.CompletedProcess[bytes]:
         stream=sys.stderr,
     )
 
-    uv_available = (
-        any(prefix in sys.base_prefix for prefix in ["uv/python", "uv\\python"])
-        and shutil.which("uv") is not None
-    )
+    env: dict[str, str] = {
+        **os.environ,
+        "UV_PYTHON": sys.executable,
+        "UV_PYTHON_DOWNLOADS": "never",
+        "FORCE_COLOR": "1",
+        "CLICOLOR_FORCE": "1",
+    }
+    uv_index_url = env.get("UV_INDEX_URL", env.get("PIP_EXTRA_INDEX_URL"))
+    if uv_index_url:
+        env["UV_INDEX_URL"] = uv_index_url
 
-    if uv_available:
-        pip_args = ["uv", "pip", "install"]
+    uv: str | None = shutil.which("uv")
```

Comment:
The issue @malfet reported is that the `--user` flag we use below is not supported by uv today.
We should move the logic to enable that flag above here and not enable uv when this flag is set.

Reply:
Updated.